src/tokens/tokens.service.ts

```diff
@@ -129,101 +156,156 @@ export class TokensService {
     return requestOptions;
   }
 
-  async createPool(dto: TokenPool): Promise<AsyncResponse> {
+  async createPool(dto: TokenPool): Promise<TokenPoolEvent> {
+    // Type must be 'fungible' or 'nonfungible'
+    if (!Object.values(TokenType).includes(dto.type)) {
+      throw new HttpException('Type must be fungible or nonfungible', HttpStatus.NOT_FOUND);
+    }
+    // Confirm that contract exists on chain
     const response = await lastValueFrom(
-      this.http.post<EthConnectAsyncResponse>(
-        `${this.instanceUrl}/create`,
-        {
-          data: encodeHex(dto.data ?? ''),
-          name: dto.name,
-          symbol: dto.symbol,
-        },
-        this.postOptions(dto.operator, dto.requestId),
-      ),
+      this.http.get<EthConnectContractsResponse>(`${this.baseUrl}/contracts/${dto.config.address}`),
```

Comment:
Should confirm that this works even if the contract was deployed outside of ethconnect.

Reply:
Removed